core/src/main/java/io/atomix/core/impl/ClasspathScanningAtomixRegistry.java

```diff
@@ -43,25 +44,26 @@ public ClasspathScanningAtomixRegistry(ClassLoader classLoader, Class<? extends
 
   public ClasspathScanningAtomixRegistry(ClassLoader classLoader, Collection<Class<? extends NamedType>> types) {
     final String scanSpec = System.getProperty("io.atomix.scanSpec");
-    final FastClasspathScanner classpathScanner = scanSpec != null ?
-        new FastClasspathScanner(scanSpec).addClassLoader(classLoader) :
-        new FastClasspathScanner().addClassLoader(classLoader);
+    final ClassGraph classGraph = scanSpec != null ?
+            new ClassGraph().enableClassInfo().whitelistPackages(scanSpec).addClassLoader(classLoader) :
```

Comment:
This will work, as long as scanSpec doesn't start with `'-'` (for blacklisting) or `'!'` (for scanning system jars), as was the case for FastClasspathScanner. Also, this form only lets you scan one whitelisted package. Maybe you want to support File.pathSeparatorChar as a separator for different packages to scan or something, then split scanSpec into an array for passing into `whitelistPackages()`?

Reply:
I'll add support for multiple packages, perhaps not with pathSepChar as it's platform dep.